### PR TITLE
docs: default_tags contribution guidelines and resource support

### LINKED ISSUE #7926: Default tags for all resources
When creating service stacks on AWS with Terraform, I like to tag all resources of a project. Right now, this involves a lot of manual work.
I would like to set default tags with the provider, that then automatically get applied to all AWS resources that support it.

There has been an [issue over on the main terraform repo (since 2015)](https://github.com/hashicorp/terraform/issues/2283) which had plenty of 👍 and got closed in favor of creating issues with the respective providers. I couldn't find such an issue though, thus creating this one.

### New or Affected Resource(s)
* `aws_*`

### Potential Terraform Configuration
```hcl
provider "aws" {
  # ...

  default_tags = {
    Project = "Project_1"
    Environment = "Production"
  }
}
```

### References
* Original issue: hashicorp/terraform#2283

---

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #7926

Output from acceptance testing:

```
N/A
```

### CODE REVIEW

File: docs/contributing/contribution-checklists.md [hunk 1]
```diff
@@ -281,80 +284,139 @@ More details about this code generation, including fixes for potential error mes
...
+  }
   ```
 
 - In the resource `Update` function (this may be the first functionality requiring the creation of the `Update` function), implement the logic to handle tagging updates, e.g. with EKS Clusters:
 
   ```go
-  if d.HasChange("tags") {
-    o, n := d.GetChange("tags")
+  if d.HasChange("tags_all") {
+    o, n := d.GetChange("tags_all")
```
Comment (ewbankkit) [inline docs/contributing/contribution-checklists.md:419]: Do we want a note on having to use `d.HasChangesExcept("tags", "tags_all")` if currently using `d.HasChangesExcept("tags")`?

Comment (anGie44) [reply docs/contributing/contribution-checklists.md:419]: definitely 👍 will add!

Comment (anGie44) [reply docs/contributing/contribution-checklists.md:419]: added in https://github.com/hashicorp/terraform-provider-aws/pull/19084/commits/a5e338530f680f2b28a274ae6c1e2c987ca34206

File: docs/contributing/contribution-checklists.md [hunk 2]
```diff
@@ -241,18 +241,21 @@ There are also functions `naming.TestCheckResourceAttrNameWithSuffixGenerated` a
...
 
 Some AWS components support [resource-based IAM policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_identity-vs-resource.html) to control permissions. When implementing this support in the Terraform AWS Provider, we typically prefer creating a separate resource, `aws_{SERVICE}_{THING}_policy` (e.g. `aws_s3_bucket_policy`). See the [New Resource section](#new-resource) for more information about implementing the separate resource and the [Provider Design page](provider-design.md) for rationale.
+
 ## Adding Resource Tagging Support
 
 AWS provides key-value metadata across many services and resources, which can be used for a variety of use cases including billing, ownership, and more. See the [AWS Tagging Strategy page](https://aws.amazon.com/answers/account-management/aws-tagging-strategies/) for more information about tagging at a high level.
 
-Implementing tagging support for Terraform AWS Provider resources requires the following, each with its own section below:
+As of version 3.38.0 of the Terraform AWS Provider, resources that previously implemented tagging support via the argument `tags`, now support provide-wide default tagging.
```
Comment (ewbankkit) [inline docs/contributing/contribution-checklists.md:249]:
```suggestion
As of version 3.38.0 of the Terraform AWS Provider, resources that previously implemented tagging support via the argument `tags`, now support provider-wide default tagging.
```

File: docs/contributing/contribution-checklists.md [hunk 3]
```diff
@@ -281,89 +284,165 @@ More details about this code generation, including fixes for potential error mes
...
+    if _, err := conn.CreatePolicyVersion(request); err != nil {
+        return fmt.Errorf("error updating IAM policy %s: %w", d.Id(), err)
+    }
+  }
+  ```
+
 ### Resource Tagging Acceptance Testing Implementation
 
-- In the resource testing (e.g. `aws/resource_aws_eks_cluster_test.go`), verify that existing resources without tagging are unaffected and do not have tags saved into their Terraform state. This should be done in the `_basic` acceptance test by adding a line similar to `resource.TestCheckResourceAttr(resourceName, "tags.%s", "0"),`
+- In the resource testing (e.g. `aws/resource_aws_eks_cluster_test.go`), verify that existing resources without tagging are unaffected and do not have tags saved into their Terraform state. This should be done in the `_basic` acceptance test by adding one line similar to `resource.TestCheckResourceAttr(resourceName, "tags.%s", "0"),` and one similar to `resource.TestCheckResourceAttr(resourceName, "tags_all.%s", "0"),`
```
Comment (ewbankkit) [inline docs/contributing/contribution-checklists.md:445]:
```suggestion
- In the resource testing (e.g. `aws/resource_aws_eks_cluster_test.go`), verify that existing resources without tagging are unaffected and do not have tags saved into their Terraform state. This should be done in the `_basic` acceptance test by adding one line similar to `resource.TestCheckResourceAttr(resourceName, "tags.%s", "0"),` and one similar to `resource.TestCheckResourceAttr(resourceName, "tags_all.%", "0"),`
```

**Review by ewbankkit — APPROVED** [review]
Other than minor suggested changes, LGTM 🚀.

**Pull request merged**